src/com/page/event/mygrades/mygrades.js

```diff
@@ -0,0 +1,101 @@
+import {h, Component} from 'preact/preact';
+
+import ContentSimple					from 'com/content-simple/simple';
+
+import ContentList				from 'com/content-list/list';
+import GradedGame				from 'com/content-event/event-graded-game';
+import ContentLoading			from 'com/content-loading/loading';
+
+import $Grade					from '../shrub/js/grade/grade';
+import $Node					from '../shrub/js/node/node';
+
+export default class MyGrades extends Component {
+
+	constructor(props) {
+		super(props);
+		this.state ={
+			'loading': true,
+		};
+	}
+	componentDidMount() {
+		$Grade.GetGameList(this.props.node.id)
+			.then(r => {
+				this.setState({'gameIds': r.games, 'nodes': new Map(), 'loading': true, 'error': null});
+				this.collectNodes(r.games);
+			})
+			.catch(r => {
+				this.setState({'error': r, 'gameIds': null, 'loading': false});
+			});
+	}
+
+	collectNodes(gameIds) {
+		let promises = [];
+		let mapping = new Map();
+		const chunkSize = 20;
+		for (let i = 0; i < gameIds.length; i += chunkSize) {
+			promises.push($Node.Get(gameIds.slice(i, i + chunkSize))
+				.then(r => {
+					r.node.map(node => {
+						mapping[node.id] = node;
+					});
+				})
+				.catch(r => {
+					this.setState({'error': r});
+				})
+			);
+		}
+
+		Promise.all(promises)
+			.then(r => {
+				this.setState({'loading': false, 'nodes': mapping});
+			});
+	}
+
+    render( props, {gameIds, error, nodes, loading} ) {
```

Comment:
```js
render( props, state ) {
    let {gameIds, error, nodes, loading} = state;
```

Reply:
Fixed, but using `const` as that is what they are.